### PR TITLE
Bump Monaco to 0.18.1

### LINKED ISSUE #176: migrate to monaco 0.18.x

### PR DESCRIPTION
**Pull request by PHameete**
Closes #176 

Updated Monaco to 0.18.1.

### CODE REVIEW

File: client/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "monaco-languageclient",
-    "version": "0.10.2",
+    "version": "0.10.3",
```
Comment (akosyakov) [inline client/package.json]: please don't do it manually, version is updated automatically during the release

File: yarn.lock [hunk 2]
```diff
@@ -4474,6 +4474,11 @@ monaco-editor-core@^0.17.0:
   resolved "https://registry.yarnpkg.com/monaco-editor-core/-/monaco-editor-core-0.17.0.tgz#8d2b7f7e02ac68f68eecc881d7a26457cbc09b49"
```
Comment (akosyakov) [inline yarn.lock]: `monaco-editor-core@^0.17.0` should be removed, could you check why it is pulled, i.e. with `yarn why monaco-editor-core@0.17.0`

File: client/src/monaco-languages.ts [hunk 3]
```diff
@@ -138,7 +138,7 @@ export class MonacoLanguages implements Languages {
                     triggerKind: 1,
```
Comment (akosyakov) [inline client/src/monaco-languages.ts]: we should be able to pass real context now

File: client/src/monaco-languages.ts [hunk 4]
```diff
@@ -128,7 +128,7 @@ export class MonacoLanguages implements Languages {
         // TODO support regrigger characters after Monaco udpate
```
Comment (akosyakov) [inline client/src/monaco-languages.ts]: monaco should be able to accept regtrigger characters now

File: client/package.json (hunk 1 above)
Comment (PHameete) [reply client/package.json]: Does this also count for the version number used by the examples?

Comment (PHameete) [reply client/package.json]: I'll just put all the versions for the languageclient back to 0.10.2 and leave it up to the merger / the release script to bump them.

**Comment by PHameete** [comment]
@akosyakov I've added the changes for the two TODOs as well.

**Comment by PHameete** [comment]
@akosyakov I'll be on vacation for a week. Could you take a look at this in the meantime? When I'm back it would be cool if we can test the new dev release or fix any potential issues that remain. Cheers!

**Comment by akosyakov** [comment]
@PHameete Could you make sure that you have this check on: https://help.github.com/en/articles/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests? It will allow us to fix any issues with additional commits to this PR and then merge it. Thank you!

**Comment by PHameete** [comment]
@akosyakov it's done. Cheers!

**Comment by rcjsuen** [comment]
Do we need to update `examples/brower/package.json`?

https://github.com/TypeFox/monaco-languageclient/blob/bab5efae562129188375227d202e544a87177e9c/examples/browser/package.json#L4-L6

**Comment by PHameete** [comment]
After the language client version is incremented , yes.

**Comment by akosyakov** [comment]
@rcjsuen it happens automatically during release: https://github.com/TypeFox/monaco-languageclient/blob/bab5efae562129188375227d202e544a87177e9c/package.json#L23

**Comment by akosyakov** [comment]
I won't be able to have a look in October. If someone can review and merge, go ahead. After merging on master it will be available under `next` tag. I also published it under `dev` tag for downstream testing.

**Comment by PHameete** [comment]
I've tested the dev release for our DSL editor. For us all existing behavior are still working correctly, and quickfix is now working as well.
We are not using the SignatureHelpContext or retrigger characters so I was not able to test those two additions.

**Comment by mhintzke13** [comment]
+1

@PHameete Oh wow thank you for doing this!

I just spent the last 2 hours diving into all the code trying to figure out why the runtime was blowing up and this PR is 100% the fix to all of the problems I'm having. I would love to get this bumped and reviewed ASAP as well, otherwise I'm going to have to cherry-pick your build into my app.

**Comment by PHameete** [comment]
@mhintzke13 this version is already published on npm as version [0.10.3-dev.3](https://www.npmjs.com/package/monaco-languageclient/v/0.10.3-dev.3)
Its just not reviewed/battle tested but you can try it out. I'm using this version myself and have had no issues so far. I don't use the full feature set of monaco though.

**Comment by mhintzke13** [comment]
@PHameete oh ok awesome, that will make this much easier to use then.  Our application is still in its infancy, so I presume everything will work just fine once these runtime exceptions are fixed. I will give it a shot and let you know how things turn out. Thanks again!

**Comment by mhintzke13** [comment]
@PHameete Everything seems to be working out pretty well so far except for this one issue.  I needed to hook into the MonacoLanguageClient middleware so I could handle Diagnostics in our application. However, when I tried to do so, the typings from `vscode` aren't available to the application.

This is the signature that is inside this file:

https://github.com/microsoft/vscode-languageserver-node/blob/153fa6316ab8991dfae25836780b66217f92efe2/client/src/client.ts#L448

And this is how I am trying to use it:

image

As you can see, I don't have access to the `Diagnostic` and `Uri` types from 'vscode'.  Have you worked around that at all yet? I'm not sure what the consequences would be if I tried to install 'vscode'.  I don't think this is directly related to this PR, so I can move this comment to Issues if it makes more sense.

**Comment by PHameete** [comment]
@akosyakov can you review this if you have some time for it?

**Comment by CGNonofr** [comment]
We use it in production for several week and everything works fine so far!

**Comment by gins3000** [comment]
Any updates with this, or help needed?

**Comment by PHameete** [comment]
Waiting for a maintainer to verify and publish. I guess TypeFox has different priorities currently. Is it possible that more maintainers are added for this project?

**Comment by gins3000** [comment]
> @mhintzke13 this version is already published on npm as version [0.10.3-dev.3](https://www.npmjs.com/package/monaco-languageclient/v/0.10.3-dev.3)
> Its just not reviewed/battle tested but you can try it out. I'm using this version myself and have had no issues so far. I don't use the full feature set of monaco though.
Thanks! I have upgraded to this version myself now, and it looks to be working nicely (just to add another +1 voice to those who are using it already).

**Comment by apupier** [comment]
@PHameete the branch is marked as "out-of-date with the base branch"

**Comment by johnwiseheart** [comment]
According to @akosyakov this should end up under the `next` tag now?

**Comment by akosyakov** [comment]
> According to @akosyakov this should end up under the next tag now?

yes

I'v also just published it under `latest` as 0.11.0

**Comment by PHameete** [comment]
Thank you @akosyakov

**Pull request merged**